lib/ansible/modules/cloud/amazon/aws_dynamodb_item.py

```diff
@@ -0,0 +1,273 @@
+- name: Deletes arribute 'person'
+  aws_dynamodb_item:
+    table: narcos
+    state: present
+    primary_key: {"bank": {"S": "hsbc"}, "quantity": {"N": "1000"}}
+    update_expression: 'REMOVE person'
+
+# Connects with awscli profile 'development' and
+# deletes a single item from 'releases' DynamoDB table
+# where the column 'project' equals string 'potatoes' (also primary key)
+# and column 'version' equals numeric '123456'
+- name: Delete a single item
+  aws_dynamodb_item:
+    profile: development
+    table: releases
+    state: absent
+    primary_key: {"project": {"S": "potatoes"}}
+    condition_expression: version = :number
+    expression_attribute_values: {":number": {"N": "123456"}}
+'''
+
+RETURN = '''
+response_metadata:
+  description: aws response metadata
+  returned: success
+  type: dict
+  sample:
+    http_headers:
+      content-length: 1490
+      content-type: text/xml
+      date: Tue, 07 Feb 2017 16:43:04 GMT
+      x-amz-crc32: 2745614147
+      x-amzn-requestid: 7f436dea-ed54-11e6-a04c-ab2372a1f14d
+    http_status_code: 200
+    request_id: 7f436dea-ed54-11e6-a04c-ab2372a1f14d
+    retry_attempts: 0
+'''
+
+from ansible.module_utils.ec2 import camel_dict_to_snake_dict
+from ansible.module_utils.aws.core import AnsibleAWSModule
+
+try:
+    import botocore
+except ImportError:
+    pass    # Handled by AnsibleAWSModule
+
+
+def update(connection, module, response, **params):
+
+    try:
+        if not module.check_mode:
+            response = connection.update_item(**params)
+        changed = True
+        return response, changed
+    except (botocore.exceptions.ClientError, botocore.exceptions.BotoCoreError) as e:
+        if e.response['Error']['Code'] == 'ResourceNotFoundException':
+            module.fail_json_aws(e, msg="Table {0} doesnt exist".format(
+                module.params.get('table')))
+        if e.response['Error']['Message'] == 'The provided key element does not match the schema':
+            module.fail_json_aws(
+                e, msg="Check the primary key, it doesnt match your table config")
+        if 'Attribute name is a reserved keyword' in e.response['Error']['Message']:
+            module.fail_json_aws(
+                e, msg='''You are trying to use a DynamoDB reserved word,
+                see https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/ReservedWords.html
+                Use expression_attribute_names to deal with this case.''')
+        if 'contains invalid key' in e.response['Error']['Message']:
+            module.fail_json_aws(
+                e, msg='''You are trying to use a DynamoDB reserved word,
+                see https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/ReservedWords.html
+                Use expression_attribute_names to deal with this case.''')
+        else:
+            module.fail_json_aws(e, msg="Error")
+
+
+def delete(connection, module, response, **params):
```

Comment:
Several things might need to be changed
  - check mode should reflect changed accurately
  - unless get_item can check condition expressions when that option is provided get_item shouldn't be used for idempotent behavior
  - it looks like the parameters for get_item and delete_item are different

I think something like this addresses all three concerns:
```suggestion
def delete(connection, module, response, **params):

    changed = True
    if not params.get('ConditionExpression'):
        get_item_param_keys = ['TableName', 'Key', 'ExpressionAttributeNames']
        get_item_params = dict((k, v) for k, v in params.items() if k in get_item_param_keys)
        try:
            response = connection.get_item(**get_item_params)
            if not response.get('Item'):
                changed = False
        except (botocore.exceptions.ClientError, botocore.exceptions.BotoCoreError) as e:
            module.fail_json_aws(e, msg="Failed to get item {0} from table {1}".format(
                params.get('TableName'), params.get('Key')))
     try:
         if not module.check_mode:
             response = connection.delete_item(**params)
         return response, changed
    except (botocore.exceptions.ClientError, botocore.exceptions.BotoCoreError) as e:
        # your exception handling
```